src/apis/impl/NotificationApiImpl.ts

```diff
@@ -97,4 +94,27 @@ export class NotificationApiImpl implements NotificationApi {
     });
   }
 
+  async concatenateFilterResults(filter: Filter<Notification>, nextId?: string | undefined, limit?: number | undefined, accumulator: Array<Notification> = []): Promise<Array<Notification>> {
+    const paginatedNotifications = await this.filterNotifications(filter, nextId, limit);
+    return !paginatedNotifications.nextKeyPair?.startKeyDocId
+      ? accumulator.concat(paginatedNotifications.rows)
+      : this.concatenateFilterResults(filter, paginatedNotifications.nextKeyPair.startKeyDocId, limit, accumulator.concat(paginatedNotifications.rows))
+  }
+
+  async getPendingNotificationsFromNewUsers(): Promise<Array<Notification>> {
+    const user = await this.userApi.getCurrentUser();
+    if (!user) throw new Error("There is no user currently logged in");
+    if (!user.healthcarePartyId) throw new Error("User is not a Healthcare Party");
+    const filter = await new NotificationFilter()
+      .forHcParty(user.healthcarePartyId)
+      .withType(NotificationTypeEnum.NEW_USER_OWN_DATA_ACCESS)
```

Comment:
Take all type of notifications for the current user, he will treat them one by one afterwards

Reply:
We have a problem for this one, as there isn't a Filter to get all the notification for a single HCP, so if we remove the type we will get all the notifications for all the HCP

Reply:
True : I'm adding a healthcarePartyId field in the afterDate filter (which is even mandatory, to avoid users to have notifications of their whole group) so that we can have a quick fix for this